File: ios/Runner/MobilePlugin.swift
import UIKit
import Flutter
import Mobile

public class MobilePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    // Note: In release 2.10, the Task Queue API is only available on the master channel for iOS.
    // let taskQueue = registrar.messenger.makeBackgroundTaskQueue()
    // let channel = FlutterMethodChannel(name: "kubenav.io", binaryMessenger: registrar.messenger(), codec: FlutterStandardMethodCodec.sharedInstance, taskQueue: taskQueue)
    let channel = FlutterMethodChannel(name: "ricoberger.de/go-flutter", binaryMessenger: registrar.messenger())
    let instance = MobilePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    if call.method == "sayHi" {
      if let args = call.arguments as? Dictionary<String, Any>,
        let name = args["name"] as? String
      {
        sayHi(name: name, result: result)
      } else {
        result(FlutterError(code: "BAD_ARGUMENTS", message: nil, details: nil))
      }
    } else if call.method == "sayHiWithDuration" {
      if let args = call.arguments as? Dictionary<String, Any>,
        let name = args["name"] as? String,
        let duration = args["duration"] as? String
      {
        sayHiWithDuration(name: name, duration: duration, result: result)
      } else {
        result(FlutterError(code: "BAD_ARGUMENTS", message: nil, details: nil))
      }
    } else {
      result(FlutterMethodNotImplemented)
    }
  }

  private func sayHi(name: String, result: FlutterResult) {
    var error: NSError?

    let data = MobileSayHi(name, &error)
    if error != nil {
      result(FlutterError(code: "SAY_HI_FAILED", message: error?.localizedDescription ?? "", details: nil))
    } else {
      result(data)
    }
  }

  private func sayHiWithDuration(name: String, duration: String, result: FlutterResult) {
    var error: NSError?

    let data = MobileSayHiWithDuration(name, duration, &error)
    if error != nil {
      result(FlutterError(code: "SAY_HI_WITH_DURATION_FAILED", message: error?.localizedDescription ?? "", details: nil))
    } else {
      result(data)
    }
  }
}

File: android/app/src/main/kotlin/de/ricoberger/goflutter/goflutter/MobilePlugin.kt
package de.ricoberger.goflutter.goflutter

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.*
import io.flutter.plugin.common.MethodChannel.*
import java.lang.Exception
import java.util.concurrent.Executors;

import mobile.Mobile;

class MobilePlugin : FlutterPlugin, MethodCallHandler {
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    val taskQueue = flutterPluginBinding.binaryMessenger.makeBackgroundTaskQueue()
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "ricoberger.de/go-flutter", StandardMethodCodec.INSTANCE, taskQueue)
    channel.setMethodCallHandler(this)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }

  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
    if (call.method == "sayHi") {
      val name = call.argument<String>("name")

      if (name == null) {
        result.error("BAD_ARGUMENTS", null, null)
      } else {
        sayHi(name, result)
      }
    } else if (call.method == "sayHiWithDuration") {
      val name = call.argument<String>("name")
      val duration = call.argument<String>("duration")

      if (name == null || duration == null) {
        result.error("BAD_ARGUMENTS", null, null)
      } else {
        sayHiWithDuration(name, duration, result)
      }
    } else {
      result.notImplemented()
    }
  }

  private fun sayHi(name: String, result: MethodChannel.Result) {
    try {
      val data: String = Mobile.sayHi(name)
      result.success(data)
    } catch (e: Exception) {
      result.error("SAY_HI_FAILED", e.localizedMessage, null)
    }
  }

  private fun sayHiWithDuration(name: String, duration: String, result: MethodChannel.Result) {
    try {
      val data: String = Mobile.sayHiWithDuration(name, duration)
      result.success(data)
    } catch (e: Exception) {
      result.error("SAY_HI_WITH_DURATION_FAILED", e.localizedMessage, null)
    }
  }
}
